src/graph.ts

```diff
@@ -902,8 +902,13 @@ export class Graph {
             }
     
             if(visitedEdgeList.length > 0) {
-                var startPoint = turfHelpers.point(feature.geometry.coordinates[0]);
-                var endPoint = turfHelpers.point(feature.geometry.coordinates[feature.geometry.coordinates.length - 1]);
+                // Use the first non-null point (in the matches), rather than just the first point from the raw trace
+                // This way, we are guaranteed to get a match if the OSRM matching was successful
+                var firstNonNull = 0, lastNonNull = matches['tracepoints'].length - 1
+                while (!matches['tracepoints'][firstNonNull] && firstNonNull++);
+                while (!matches['tracepoints'][lastNonNull] && lastNonNull--);
+                var startPoint = turfHelpers.point(matches['tracepoints'][firstNonNull]['location']);
+                var endPoint = turfHelpers.point(matches['tracepoints'][lastNonNull]['location']);
```

Comment:
The start and end point of the journey was previously being found by "snapping" the raw gps to the matched output from OSRM. However, this was sometimes outside the `searchRadius` if OSRM had failed to match the first couple of points, or there is large GPS noise.

Why match again, when OSRM has already mapped the raw points to the route? Instead now, I just grab the first non-null match for the start point, and the final non-null match for the end point.